Osx tweaks

mveytsman opened the pull request:
I tweaked the osx keybinding.

1) The osx bindings used the emacs version of undo, redo, copy, and paste. I switched to the evil equivalents. I'm assuming this is within the spirit of spacemacs.
2) I added `s-n` for new frame, `s-W` for close frame, and `s-a` for select all.

mveytsman commented:
Now that i think about it, a more accurate `s-w` keybinding would be `delete-window` rather then `kill-this-buffer`, but I don't want to change an existing keybinding willy-nilly

trishume commented on contrib/osx/keybindings.el:
```diff
@@ -7,14 +7,17 @@
+  (global-set-key (kbd "s-a") 'mark-whole-buffer)
   (global-set-key (kbd "s-x") 'kill-region)
   (global-set-key (kbd "s-w") 'kill-this-buffer)
-  (global-set-key (kbd "s-z") 'undo-tree-undo)
```
This is wrong, spacemacs uses a special undo thingy that has fancy tree-ness. Try pressing `SPC a u`. The undo and redo bindings should not be changed.

mveytsman replied:
Thanks for reviewing my PR @trishume.

From what I understand, spacemacs sets `global-undo-tree-mode` (https://github.com/syl20bnr/spacemacs/blob/develop/spacemacs/packages.el#L2695), so regular `undo` behaves like `undo-tree-undo`. 

I personally disable undo-tree-mode in my spacemacs, so undo behaves as expected. 

Setting the `s-z` keybinding to `undo` will make it behave as expected regardless of whether you like undo-tree-mode

Thanks!

trishume replied:
This is interesting, reading the source it looks like `undo-tree-mode` doesn't redefine the `undo` function, it switches the keybindings to `undo-tree-undo'. However the default spacemacs `u` key is bound to just `undo`.

Reading the source of `undo-tree-undo` it looks like it it only makes a difference if you are in the undo-tree visualization. I think this is a bug in spacemacs and if you hit `u` while visiting a previous state in the undo tree it will wreck stuff.

Anyhow I think since Spacemacs defaults to `undo-tree-mode` it should do that in the keybindings, or call a special function that decides what to do based on if `undo-tree-mode` is enabled. You can always set them to the plain functions in your dotfile.

trishume commented:
This is good, agreed about `s-w` for delete-window. Just revert the change to the undo bindings, maybe add the `s-w` binding and this is great :+1:

mveytsman commented:
Okay, I reverted the `undo-tree-undo` binding and changed `s-w` to `delete-window`

Does having 543c624 in this PR mess up things on your end? Let me know if you need me to rebase and squash all these commits @trishume.

Cheers,

trishume commented:
1. You should really rebase on develop instead of merging, keeps the history clean. I'm not git pro enough to know how easy it would be to undo the merge you already did.
2. `s-w` isn't documented in the readme
3. redo is still on the default one instead of the undo tree one.

mveytsman commented:
1) Done
2) The readme documented it as "Close window" which I left. I think it's accurate to what's it doing now (calling `delete-window`), but this runs into the whole problem of emacs nomenclature for frames & windows.
3) Fixed.

syl20bnr commented:
Thank you ! :+1: 
Cherry-picked into `develop` branch, you can safely delete your branch.
